### DISCUSSION TITLE
Chemprop & continual learning

### DISCUSSION THREAD

**Discussion by Rhys-McAlister** [chemprop/chemprop#962] [Q&A]
Hi all, I've been trying to / thinking about implementing a couple of continual  learning techniques into model and was wondering if anyone had any experience approaching this issue.

My setup so far is having an ensemble of models, quantifying the uncertainty of the predictions using the ensemble predictions to flag poorly performing samples and then generating new data for these samples. I would prefer to be able to use continual learning as opposed to just re-training the entire model. 

Any thoughts would be greatly appreciated.

Cheers

**Comment by KnathanM** [comment]
Hi @Rhys-McAlister, sorry I didn't see your post. Did you find something that worked for you? I expect you could the continual learning in a jupyter notebook, just by making a new dataloader with the new data samples and a new lightning trainer with the number of epochs set as you'd like, every time you sample new data. This could bias your model to the newer data though.

**Reply by Rhys-McAlister** [reply]
Hi @KnathanM , no problem. I had a look over the literature for continual learning to try and combat exactly that issue of biasing towards more recent data points but wasn't able to find anything super concrete on continual learning over graphs for molecular property prediction. There was an interesting idea on selecting random samples from prior training data to interleave into the new batches of continual learning training data but I wasn't able to produce any meaningful results from this.

I was thinking perhaps it might just be a better idea to have a large pretrained model and simply add the extra data points to the fine-tuning dataset and do this in cycles.

If you had any ideas around implementing CL, active learning or anything similar I'd be glad to hear.
